docs(consul-k8s): add note about config entries requiring transparent proxy on both downstream and upstream (#2006)
hashicorp/web-unified-docs: content/consul/v1.18.x/content/docs/connect/config-entries/proxy-defaults.mdx

Contains configurations for proxies that are running in transparent proxy mode. This mode enables permissive mTLS for Consul so that you can use your Kubernetes cluster's DNS service instead of Consul DNS. Refer to [Transparent proxy mode](/consul/docs/k8s/connect/transparent-proxy) for additional information.

> **Note**: If you plan to use config entries to define a call path (for example, `ServiceDefaults` or `ProxyDefaults` that affect upstream routing), enable transparent proxy on both the downstream and upstream services. If an upstream must bypass the mesh, do NOT apply config entries that assume a mesh-based call path to that upstream.

Reviewer: ```suggestion
When you use configuration entries to define a call path that affects upstream routing, you must enable transparent proxy on both the downstream and upstream services. If an upstream must bypass the mesh, do not apply configuration entries such as `ServiceDefaults` or `ProxyDefaults`, which assume a mesh-based call path to that upstream.
```
Suggestions for style guide alignment.
- The Note callout is not recommended for this use case.
- Avoid parenthesis
- Do not use capital letters for emphasis.
